Source/Scene/EllipsoidPrimitive.js

```diff
@@ -398,7 +423,10 @@ define([
             }
 
             // Recompile shader when material changes
-            if (materialChanged || lightingChanged || !defined(this._pickSP)) {
+            if (materialChanged || lightingChanged || !defined(this._pickSP) || useLogDepthChanged) {
```

Comment:
> `useLogDepthChanged`

If it makes things much easier, I don't know that users will want to be able to toggle log depth at runtime.